rpcserver.go

```diff
@@ -1185,18 +1186,66 @@ func prepareAndSubmitOrder(ctx context.Context, o order.Order,
 	)
 }
 
+// lndVersionIsCompatible makes sure the detected lnd version is compatible
+// with our current version requirements. True is returned if the version is
+// compatible, and false otherwise.
+func lndVersionIsCompatible(actual *verrpc.Version,
+	expected *verrpc.Version) bool {
+
+	// We need to check the versions parts sequentially as they are
+	// hierarchical.
+	if actual.AppMajor != expected.AppMajor {
```

Comment:
BTW linter is complaining about not using the pattern

```
if actual.AppMajor != expected.AppMajor {
    return actual.AppMajor > expected.AppMajor
}
```

Reply:
Should be gone now that the newly exported `lndclient` method is used instead.